### PR TITLE
Add Accordian for similar projects

### LINKED ISSUE #289: Improve project page design
the distribution of the things in the [project page](https://hackweek.suse.com/17/projects/jetpad-online-collaborative-text-editor) looks strange:

image

Too empty, and the projects are in side really compressed.

I think we can find a better design for this page. :wink: Maybe it is a good idea to have at the button a dedicated section for the related projects.

### PR DESCRIPTION
**Pull request by AnkushMalik**
Add accordian for similar project in projects#show
Closes #289

image

image

### CODE REVIEW

File: spec/models/project_spec.rb [hunk 1]
```diff
@@ -129,8 +129,8 @@
       project2.add_keyword! 'web', user
     end
 
-    it 'provides array of projects having similar tags' do
-      expect(project.similar_projects).to include(project2)
+    it 'provides array of common keywords in projects' do
+      expect(project.similar_projects_keywords).to include(Keyword.first)
```
[the remark this answers is not in the document]
Comment (AnkushMalik) [reply spec/models/project_spec.rb]: > We should also test the case for when there are no keywords
ok 👍
> I'd go with .to include('web')
line:133 is checking for Keyword entry not the keyword.name string 
i can change it to ```.to include(Keyword.find_by_name('web')) ``` if you like :)

**Comment by AnkushMalik** [comment]
> The project buttons could go above the hr

Iam not clear about hr are you talking about the border-bottom of '.page-header.media'

> Moreover similar projects are on the same line as Comments, and below their box there is a help button which refers to Comments, not similar projects

I agree , what should i do about it then? suggestion 😅 ? pull-left?

**Comment by differentreality** [comment]
hackweek_project_buttons

**Comment by differentreality** [comment]
Regarding the layout I am thinking something like the following, but I would expect to have some design we, more or less, agree on, prior to start coding the changes.

hackweek_project_show_layout

File: app/views/projects/_similar_projects.html.haml [hunk 2]
```diff
@@ -1,8 +1,20 @@
...
+        %h4.panel-title
+          %a{'aria-controls': "collapse_#{i}", 'aria-expanded': 'true', 'data-parent': '#accordion', 'data-toggle': 'collapse', href: "#collapse_#{i}", role: 'button'}
+            = word.name
+            %i.fa.fa-caret-right
+      .panel-collapse.collapse{id: "collapse_#{i}", 'aria-labelledby': "heading_#{i}", role: 'tabpanel'}
+        .panel-body
+          - similar_projects = (word.projects.current(Episode.active)-[@project]).sample(5)
+          - similar_projects.each do |project|
+            %h5
+              = link_to project.title, project_path(project)
```
Comment (DavidKang) [inline app/views/projects/_similar_projects.html.haml]: missing parenthesis

File: app/views/projects/_similar_projects.html.haml [hunk 3]
```diff
@@ -1,8 +1,20 @@
...
+              by
+              = link_to project.originator.name, user_path(project.originator)
```
Comment (DavidKang) [inline app/views/projects/_similar_projects.html.haml]: missing parenthesis

File: app/views/projects/show.html.haml [hunk 4]
```diff
@@ -84,8 +84,8 @@
   .col-sm-4
     .col-sm-11#similar_projects
       %h4 Similar Projects
-      - if @similar_projects.present?
-        = render :partial => 'similar_projects', collection: @similar_projects, as: :project
+      - if @similar_projects_keys.present?
+        = render :partial => 'similar_projects'
```
Comment (DavidKang) [inline app/views/projects/show.html.haml]: please use the new sintax `render partial: 'similar_projects'`

**Pull request merged**